app/actions/ControlActions.js

```diff
@@ -409,184 +265,111 @@ export const SETMAXPRICEABSOLUTE = "SETMAXPRICEABSOLUTE";
-        if (error) {
-          hitError += error + ". ";
-        } else {
-          cfg.set("maxperblock",maxPerBlock);
-          dispatch({ maxPerBlock: maxPerBlock, type: SETMAXPERBLOCK });
-        }
-      });
-    }
-    if (hitError != "") {
-      dispatch({ error: hitError, type: SETTICKETBUYERCONFIG_FAILED });
-    } else {
-      dispatch({ success: "Ticket buyer settings have been successfully updated.", type: SETTICKETBUYERCONFIG_SUCCESS });
       dispatch(getTicketBuyerConfigAttempt());
-    }
-  };
-}
+    })
+    .catch(error => dispatch({ error, type: SETTICKETBUYERCONFIG_FAILED }));
+};
 
 export const STARTAUTOBUYER_ATTEMPT = "STARTAUTOBUYER_ATTEMPT";
 export const STARTAUTOBUYER_FAILED = "STARTAUTOBUYER_FAILED";
 export const STARTAUTOBUYER_SUCCESS = "STARTAUTOBUYER_SUCCESS";
 
-export function startAutoBuyerAttempt(passphrase, accountNum, balanceToMaintain,
-  maxFeePerKb, maxPriceRelative, maxPriceAbsolute, maxPerBlock, stakepool) {
-  var request = new StartAutoBuyerRequest();
-  request.setPassphrase(new Uint8Array(Buffer.from(passphrase)));
-  request.setAccount(accountNum);
-  request.setBalanceToMaintain(balanceToMaintain*1e8);
-  request.setMaxFeePerKb(maxFeePerKb*1e8);
-  request.setMaxPriceRelative(maxPriceRelative);
-  request.setMaxPriceAbsolute(maxPriceAbsolute*1e8);
-  request.setVotingAddress(stakepool.TicketAddress);
-  request.setPoolAddress(stakepool.PoolAddress);
-  request.setPoolFees(stakepool.PoolFees);
-  request.setMaxPerBlock(maxPerBlock);
-  return (dispatch, getState) => {
-    dispatch({
-      type: STARTAUTOBUYER_ATTEMPT,
-    });
-    const { ticketBuyerService } = getState().grpc;
-    ticketBuyerService.startAutoBuyer(request,
-      function(error, startAutoBuyerResponse) {
-        if (error) {
-          dispatch({ error, type: STARTAUTOBUYER_FAILED });
-        } else {
-          var success = "You successfully started the auto ticket buyer.";
-          dispatch({ success: success, startAutoBuyerResponse: startAutoBuyerResponse, type: STARTAUTOBUYER_SUCCESS,
-            balanceToMaintain: balanceToMaintain,
-            maxFeePerKb: maxFeePerKb*1e8,
-            maxPriceRelative: maxPriceRelative,
-            maxPriceAbsolute: maxPriceAbsolute,
-            maxPerBlock: maxPerBlock,
-          });
-          setTimeout(()=>dispatch(getTicketBuyerConfigAttempt(), 1000));
-        }
+export const startAutoBuyerAttempt = (
+  passphrase, accountNum, balanceToMaintain, maxFeePerKb, maxPriceRelative, maxPriceAbsolute,
+  maxPerBlock, stakepool
+) => (dispatch, getState) => {
+  dispatch({ type: STARTAUTOBUYER_ATTEMPT, });
+  return wallet.startAutoBuyer(
+    sel.ticketBuyerService(getState()), passphrase, accountNum, balanceToMaintain*1e8, maxFeePerKb*1e8,
+    maxPriceRelative, maxPriceAbsolute*1e8, maxPerBlock, stakepool
+  )
+    .then(startAutoBuyerResponse => {
+      dispatch({
+        success: "You successfully started the auto ticket buyer.",
```

Comment:
Don't need this anymore as the snackbar reducers show the correct message given the `STARTAUTOBUYER_SUCCESS` action